server/player/player.go

```diff
@@ -827,7 +827,11 @@ func (p *Player) Respawn() {
 	case world.End:
 		_, w = w.PortalDestinations()
 	}
-	pos := w.Spawn().Vec3Middle()
+
+	pos := p.spawnPoint.Load()
+	if pos.ApproxEqual(mgl64.Vec3{}) {
+		pos = w.Spawn().Vec3Middle()
+	}
```

Comment:
I'm a bit concerned over how this will work when the player gets moved to a different world, do you have any idea on how we can implement this in a way that it doesn't teleport players to a spawn position that is invalid in different worlds?